### PR TITLE
add active prop to carousel

### LINKED ISSUE #292: [Feature Request] Add "active" prop to  DMC Carousel
Hi,

I would like to implement a Mantine Carousel using DMC. However I could net find any attribute that lets me the set active slide using some logic or callback. I would like the carousel to alternate between two slide every couple of seconds. Is there any way to achieve this behavior?

### PR DESCRIPTION
**Pull request by AnnMarieW**
closes #292
To do:
- [x] add tests
- [x] add changelog  

Will update docs with new examples after release and [doc PR #69](https://github.com/snehilvj/dmc-docs/pull/69)

### CODE REVIEW

**Comment by AnnMarieW** [comment]
Here is a sample app:

```python
from dash import Dash, _dash_renderer, html, Input, Output
import dash_mantine_components as dmc

_dash_renderer._set_react_version("18.2.0")

app = Dash(external_stylesheets=dmc.styles.ALL)


component = dmc.Carousel(
    [
        dmc.CarouselSlide(dmc.Center("Slide-0", bg="blue", c="white", p=60)),
        dmc.CarouselSlide(dmc.Center("Slide-1", bg="blue", c="white", p=60)),
        dmc.CarouselSlide(dmc.Center("Slide-2", bg="blue", c="white", p=60)),
    ],
    id="carousel",
   # initialSlide=1,
    autoplay={"delay": 200, "stopOnMouseEnter": True, "stopOnInteraction":False},
)


app.layout = dmc.MantineProvider(
    html.Div([
        component,
        dmc.Text(id="output")
    ])
)

@app.callback(
    Output("output", "children"),
    Input("carousel", "active")
)
def update(n):
    return f"slide index {n}"


if __name__ == "__main__":
    app.run(debug=True)


```

File: tests/test_carousel.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    app = Dash(__name__, external_stylesheets=dmc.styles.ALL)
+
+    component = dmc.Carousel(
+        [
+            dmc.CarouselSlide(dmc.Center("Slide-0", bg="blue", c="white", p=60)),
+            dmc.CarouselSlide(dmc.Center("Slide-1", bg="blue", c="white", p=60)),
+            dmc.CarouselSlide(dmc.Center("Slide-2", bg="blue", c="white", p=60)),
+        ],
+        id="carousel",
+        autoplay={"delay": 500, "stopOnMouseEnter": True, "stopOnInteraction": False, "stopOnLastSnap": True},
```
Comment (alexcjohnson) [inline tests/test_carousel.py:16]: maybe disable `stopOnMouseEnter` just so people don't accidentally halt the test when running it locally?

File: src/ts/components/extensions/carousel/Carousel.tsx [hunk 2]
```diff
@@ -3,12 +3,14 @@ import { MantineSpacing, StyleProp } from "@mantine/core";
 import { BoxProps } from "props/box";
 import { DashBaseProps } from "props/dash";
 import { StylesApiProps } from "props/styles";
-import React from "react";
+import React, { useEffect, useState } from "react";
```
Comment (alexcjohnson) [inline src/ts/components/extensions/carousel/Carousel.tsx:6]: did you use `useState`?

**Review by alexcjohnson — APPROVED** [review]
💃 Very nice - just two minor comments.

File: src/ts/components/extensions/carousel/Carousel.tsx (hunk 2 above)
Comment (AnnMarieW) [reply src/ts/components/extensions/carousel/Carousel.tsx:6]: > did you use useState?


...only while messing around with thing that didn't work haha
Need to add linting and formatting to this project  in order to clean up stuff like this automatically,

**Pull request merged**